vispy/visuals/volume.py

```diff
@@ -483,7 +518,40 @@ def __init__(self, vol, clim=None, method='mip', threshold=None,
         self.relative_step_size = relative_step_size
         self.threshold = threshold if (threshold is not None) else vol.mean()
         self.freeze()
-    
+
+    def _create_texture(self, texture_format, emulate_texture, data):
+        tex_cls = TextureEmulated3D if emulate_texture else Texture3D
+
+        tex_kwargs = {}
+        if isinstance(texture_format, str) and texture_format == 'auto':
```

Comment:
These 3 tests could probably be factorized. It is made of three consecutive conditionals making it hard to see how texture format is transformed